src/geometry/euclidean/circumcenter.lean

```diff
@@ -377,12 +377,64 @@ begin
           (λ i, hr i (set.mem_univ _))).symm
 end
 
+/-- The orthogonal projection of a point `p` onto the hyperplane spanned by the simplex's points. -/
+def orthogonal_projection_span {n : ℕ} (s : simplex ℝ P n) :=
+orthogonal_projection (affine_span ℝ (set.range s.points))
```

Comment:
Where did these new lemmas come from?

Reply:
This is commit 85b9f68f, where I fixed a timeout by splitting off some repeated parts of the quite long proof `eq_or_eq_reflection_of_dist_eq`, due to some slow defeq checks. This will be alleviated by #11759, which makes defeq easier so we don't need quite as much splitting.

Reply:
Can you split that commit to a standalone PR? For one, that will make it much easier to revert after 11759, if we feel so inclined.

Reply:
#12209